src/util/cache.h

```diff
@@ -0,0 +1,45 @@
+#pragma once
+
+#include <set>
+#include <util/algorithms.h>
+
+// Helper definitions
+using crc_hash = size_t; // from irep2_meta_templates.h
+/**
+ * @brief This class stores all asserts conditions and guards
+ *        for a given SSA, if some pair was already added
+ *        then it will convert it to be a trivial case,
+ *        e.g. GUARD => COND ----> GUARD => 1
+ */
+class crc_assert_cache : public ssa_algorithm
+{
+public:
+  using crc_pair = std::pair<crc_hash, crc_hash>;
```

Comment:
Perhaps a comment to explain the pair?
e.g. // first refers to guard_crc, second refers to cond_crc?